pkg/api/types.go

```diff
@@ -3023,6 +3023,10 @@ type NodeList struct {
 type NamespaceSpec struct {
 	// Finalizers is an opaque list of values that must be empty to permanently remove object from storage
 	Finalizers []FinalizerName
+
+	// NetworkPolicy describes the use of NetworkPolicy objects in this namespace. If not specified, then default policies will be applied and NetworkPolicy objects in the Namespace will be ignored
+	// +optional
+	NetworkPolicy *NamespaceNetworkPolicy
```

Comment:
Let me clarify layering violation so others can see train of thought here.

Namespace is the root of the Kube system.  *every* valid kubernetes ever has to deal with namespace.  By implication, the API rules are such that someone creating a namespace on one cluster can expect the cluster to respect DefaultDeny.  However - if a network plugin does not support default deny, then this field has no meaning.  A caller can't determine whether network policy is actually enforced or not (since we can't block users from creating namespaces).

Recently we've been discussing layers a lot - what is the core of the system, what are extensions, what are pieces that are extensions but still expected to always be there.  Putting something about network policy, even by implication, on namespace implies that every Kube cluster needs to at least reject / inform users that DefaultDeny won't in fact work.  That means that networking policy is now part of the core layer of Kubernetes, vs an extension that someone can use later.

I'm also very concerned about the pattern this introduces - why does NetworkPolicy get to land on Namespace and other types of policy don't?  I don't think we expected Namespace to be a bucket holder for all sorts of policy - in fact, it's actually concerning if every component that wants to add a default policy now has to do so on the namespace.

From a modeling perspective, is there a desire to limit who can set this policy?  Such that you can't set it on NetworkPolicy itself (because lower privileged users might have permission to set it)?